### DISCUSSION TITLE
Github App installation into one repo

### DISCUSSION THREAD

**Discussion by olivierodo** [community/community#24388] [Apps, API and Webhooks]
<p>Hello,<br>
I’m building a Github app and i was looking for a way to only restrict the installation of my app into a single repository.</p>
<p>My app can’t be installed into multiple repository… OR it would be very bad to allow my app to be installed on all the repository of the user.</p>
<p>Is there a way to add such a restriction?</p>
<p>Thank you.</p>

**Comment by jamesmartin** [comment, marked as answer]
<p>:wave: <a class="mention" href="/olivierodo">@olivierodo</a>!</p>
<p>There’s no way to restrict installation of your App to particular repositories via the App’s settings.</p>
<p>What’s the use-case for this? Is there a specific repository that your App <em>should</em> be installed on? Or is this something to do with billing your users for the number of repositories they install?</p>

**Reply by mfbx9da4** [reply]
The use case for me is I already know the name of the org+repo the app needs access to and I want to request access to that specific one

**Comment by olivierodo** [comment]
<p>Hello,</p>
<p>Thank you for your reply, my app should help the user to setup some base practices for testing.<br>
My plan was to push some files at the install step, however i realized that most of the people like to select all their repo (believe it or not… lol). Then since i don’t want to push files on all the user repositories, i think i will need change my flow…</p>

**Comment by jamesmartin** [comment]
<p>Interesting. So you need some way for a user to indicate that they want your App to operate on a specific repository, even though they may have installed the App on all their repos. :thinking:</p>
<p>I’m guessing you already request repository contents write permission, so maybe the trigger for your App could be the user committing a configuration file in their repository (For example, GitHub Actions are initially triggered when you upload YAML files to the <code>.github/workflows/</code> directory of a repo).</p>
<p>You could listen for the <a href="https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#push" rel="nofollow noopener"><code>push</code></a> or <a href="https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#pull_request" rel="nofollow noopener"><code>pull_request</code></a> Webhook events to see if any of the pushed files contain your App’s configuration file.</p>
